### DISCUSSION TITLE
Force turn on Threaded Optimization in osu! NVIDIA Control Panel

### DISCUSSION THREAD

**Discussion by re-vliz** [ppy/osu#25578] [Q&A]
I found a [post about Threaded Optimization on reddit](https://www.reddit.com/r/osugame/comments/7lesa0/threaded_optimization_resets_for_osu_nvidia/) and ppy answered "having it on causes stutters, so osu! forces it off".

But on my laptop, Threaded Optimization actually makes my osu! got smoother, FPS increased about 15% and more stable, input latency decreased from 4.1ms to 2.7 ms, and it rarely stutters since I enabled Threaded Optimization (because if Threaded Optimization is off, my osu! often stutters)

![Cuplikan layar 2023-11-20 084614](https://github.com/ppy/osu/assets/90372213/9a0ff007-0630-429c-b2d7-00e0242249b4)

So, my question is: 
### and why don't you guys create a switch to (force) turn on and off NVIDIA Threaded Optimization in osu! options rather than forcing it off

Thank You 😘

**Comment by re-vliz** [comment]
osu! Please don't force turn off it😭

**Comment by peppy** [comment]
This is talking about stable, right?

**Reply by DeathEnjoyer** [reply]
Yes Stable, we hope there is a way to keep it on cuz it really works well for me, or at least a way to not let it force off when the game shuts down.

**Reply by QuakedK** [reply]
it's getting to a point, I might write a paragraph or two myself. It appears everyone who has talked about Threaded Optimization gets better performance with it on, so atp why keep it disabled. It would be smarter to add it as a feature/toggle, but never in my life would I hear a game dev say your not gonna feel "Such and such lower latency" like bro do you not want you game to run better or not. I get devolopement to some degree but come on peppy lock in and don't make such a dumb take.

**Reply by bdach** [reply]
@QuakedK In lazer threaded optimisations is changed to auto so I'm not even sure what you're prattling on about.

**Reply by QuakedK** [reply]
Hey @bdach so, not everyone uses Lazer. Hope this helps, however it is useful info to know lazer addressed this.

**Reply by bdach** [reply]
https://github.com/ppy/osu/discussions/27666 then

**Reply by QuakedK** [reply]
@bdach I wasn't exactly level headed when writing my first comment, however I do now know Lazer has addressed or changed the previous behavior. But I probably won't make the switch. But the link you just sent tells the user to turn on compatibility mode. However, there's one major issue with that. When compatibility mode is on, the latency gets dramatically worse. 

This only leaves us with a few options.

Option 1:  Tediously but intentionally change your threaded optimization to on every time osu! is either restarted or your computer restarts.

Option 2: Request an intentional change to the game that allows us to either toggle or turn it on by default now.

Option 3: See if there's a way to make a script that fixes this or automates the tedious task of returning it on over and over.

Option 4: Make the switch to osu! Laser. 😔


[Compatibility Mode: On](https://github.com/user-attachments/assets/050d6b7e-098f-428d-9fd0-4dbf4fd824a1)
[Compatibility Mode: OFF](https://github.com/user-attachments/assets/505e7d26-016c-4cb0-83f1-0a932776bcf3)

**Reply by failaip12** [reply]
I wouldn't call a 0.7ms increase a dramatically worse latency, I don't think you'd notice that in a blind test. Either way a script could be easy to make, especially if you ask some AI.

**Reply by QuakedK** [reply]
@failaip12 You've never yearned for the mines of latency have you? I'd bet money top players would perform better on optimized pc's than non-optimized pc's yet yet some of the differences could be in literal nanoseconds or at or below -50ms.

Anyways I've made the auto import script -  [Nip Import](https://github.com/QuakedK/Osu-Optimizer/blob/main/Downloads/Nip%20Import.bat) and [documentation](https://github.com/QuakedK/Osu-Optimizer/tree/main#how-to-prevent-osu-from-turning-off-threaded-optimization)!

**Reply by failaip12** [reply]
I have spent way too much time looking at latency and fps and all the bullshit guides, I've seen and tried them all. Almost none of them matter, especially for osu which already runs at crazy high FPS.

I guarantee that in a blind test you wouldn't notice the difference.

**Reply by QuakedK** [reply]
@failaip12 I sort of optimize pc's for a living, and yes the average person probably couldn't tell the difference between 240hz and 360hz. But top players of any game can 9/10 times...  

And I can guarantee those fps and latency guides, you've done are nothing more than snake oil and outdated meme tweaks lmao.

**Reply by peppy** [reply]
As @failaip12 says, back it up with blind tests because we don't deal with sub-millisecond issues here (why we intentionally locked lazer to avoid this cloud chasing).

**Reply by southtowne** [reply]
> I wouldn't call a 0.7ms increase a dramatically worse latency, I don't think you'd notice that in a blind test. Either way a script could be easy to make, especially if you ask some AI.

you do realize the ""latency"" numbers shown here in game are measuring frame times, and NOT input latency, correct? meaning 0.70ms is effectively a 1,400 FPS difference. I would say ANYBODY can notice this difference.
